test/scale/scale_tester.go

```diff
@@ -117,68 +134,48 @@ scaleOutTest:
 			Replicas: int32(0),
 		},
 	}
+
+	startScaleIn := time.Now()
 	_, _ = client.AppsV1().StatefulSets(statefulSet.Namespace.Name).UpdateScale(context.TODO(), statefulSet.Statefulset.Name, newScale, metav1.UpdateOptions{})
+	numberOfRemainingPods := podutil.WaitForAllPodsWithMatchingLabelToDelete(*scaleTestTimeout, time.Minute, totalNumberOfPods, client, namespace)
+	scaleInPodsCompleted := time.Since(startScaleIn).Minutes()
 
-	numberOfRemainingPods := totalNumberOfPods
-	ticker.Reset(time.Minute)
-	tickerCount = 0
-	timeout = time.After(time.Duration(*scaleTestTimeout) * time.Minute)
-scaleInTest:
-	for {
-		select {
-		case <-timeout:
-			break scaleInTest
-		case <-ticker.C:
-			tickerCount = tickerCount + 1
-			numberOfRemainingPods = podutil.CountAllPodsWithMatchingLabel(client, namespace, map[string]string{"group": "azuredisk-scale-tester"}, string(v1.PodRunning))
-			if numberOfRemainingPods <= 0 {
-				break scaleInTest
-			}
+	numberOfRemainingVolumeAtts := volutil.WaitForAllVolumeAttachmentsToDetach(*scaleTestTimeout, 30*time.Second, totalNumberOfPods**volMountedToPod, client)
```

Comment:
Since now with the replicas we won't actually detach, should we rename this to  `WaitForAllVolumeAttachmentsToDelete`?